ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+
+    _subindex = MM_GiveNewStringWithCharacterInFront(subindex, '_');
+    sizet_subindex = strlen(_subindex);
+    sizet_nomcamp = strlen(NomCamp_SubIndex);
+
+    if (sizet_nomcamp + sizet_subindex > ampladamax - 1)
+        memcpy(NomCamp_SubIndex + ((ampladamax - 1) - sizet_subindex),
+               _subindex, strlen(_subindex));
+    else
+        NomCamp_SubIndex = strcat(NomCamp_SubIndex, _subindex);
+
+    free_function(_subindex);
+
+    return NomCamp_SubIndex;
+}
+
+MM_FIRST_RECORD_OFFSET_TYPE
+MM_GiveOffsetExtendedFieldName(const struct MM_FIELD *camp)
+{
+    MM_FIRST_RECORD_OFFSET_TYPE offset_nom_camp;
+
+    memcpy(&offset_nom_camp,
+           (char *)(&camp->reserved_2) + MM_OFFSET_RESERVAT2_OFFSET_NOM_ESTES,
+           4);
+    return offset_nom_camp;
+}
+
+int MM_WriteNRecordsMMBD_XPFile(struct MMAdmDatabase *MMAdmDB)
+{
+    GUInt32 nRecords;
+    if (!MMAdmDB->pMMBDXP || !MMAdmDB->pFExtDBF)
+        return 0;
+
+    // Updating number of features in features table
+    fseek_function(MMAdmDB->pFExtDBF, MM_FIRST_OFFSET_to_N_RECORDS, SEEK_SET);
+
+    if (MMAdmDB->pMMBDXP->nRecords > UINT32_MAX)
+    {
+        MMAdmDB->pMMBDXP->dbf_version = MM_MARCA_VERSIO_1_DBF_ESTESA;
+
+        if (fwrite_function(&MMAdmDB->pMMBDXP->nRecords, 4, 1,
+                            MMAdmDB->pFExtDBF) != 1)
+            return 1;
+    }
+    else
+    {
+        MMAdmDB->pMMBDXP->dbf_version = MM_MARCA_DBASE4;
+
+        nRecords = (GUInt32)MMAdmDB->pMMBDXP->nRecords;
+        if (fwrite_function(&nRecords, 4, 1, MMAdmDB->pFExtDBF) != 1)
+            return 1;
+    }
+
+    fseek_function(MMAdmDB->pFExtDBF, MM_SECOND_OFFSET_to_N_RECORDS, SEEK_SET);
+    if (MMAdmDB->pMMBDXP->dbf_version == MM_MARCA_VERSIO_1_DBF_ESTESA)
+    {
+        /* from 16 to 19, position MM_SECOND_OFFSET_to_N_RECORDS */
+        if (fwrite_function(((char *)(&MMAdmDB->pMMBDXP->nRecords)) + 4, 4, 1,
+                            MMAdmDB->pFExtDBF) != 1)
+            return 1;
+
+        /* from 20 to 27 */
+        if (fwrite_function(&(MMAdmDB->pMMBDXP->dbf_on_a_LAN), 8, 1,
+                            MMAdmDB->pFExtDBF) != 1)
+            return 1;
+    }
+    else
+    {
+        if (fwrite_function(&(MMAdmDB->pMMBDXP->dbf_on_a_LAN), 12, 1,
+                            MMAdmDB->pFExtDBF) != 1)
+            return 1;
+    }
+
+    return 0;
+}
+
+static MM_BOOLEAN MM_UpdateEntireHeader(struct MM_DATA_BASE_XP *data_base_XP)
+{
+    MM_BYTE variable_byte;
+    MM_EXT_DBF_N_FIELDS i, j = 0;
+    const size_t max_n_zeros = 11;
+    char *zero;
+    const MM_BYTE byte_zero = 0;
+    char ModeLectura_previ[4] = "";
+    MM_FIRST_RECORD_OFFSET_TYPE bytes_acumulats;
+    MM_BYTE name_size;
+    int estat;
+    char nom_camp[MM_MAX_LON_FIELD_NAME_DBF];
+    size_t retorn_fwrite;
+    MM_BOOLEAN table_should_be_closed = FALSE;
+    GUInt32 nRecords;
+
+    if ((zero = calloc_function(max_n_zeros)) == nullptr)
+        return FALSE;
+
```

Comment:
as max_n_zeros is small, rather than doing a dynamic allocation, you could just declare ``char zero[11] = {0};``

Reply:
Sure!Done!